tests/fe/abf_02.cc

```diff
@@ -0,0 +1,593 @@
+            boundary_functions[c] = &function;
+          project_boundary_values (dof, boundary_functions, q_boundary,
+                 boundary_values);
+        */
+      }
+
+
+  // set up mass matrix and right hand side
+  vec.reinit (dof.n_dofs());
+  SparsityPattern sparsity(dof.n_dofs(),
+                           dof.n_dofs(),
+                           dof.max_couplings_between_dofs());
+  DoFTools::make_sparsity_pattern (dof, sparsity);
+  constraints.condense (sparsity);
+
+  SparseMatrix<double> mass_matrix (sparsity);
+  Vector<double> tmp (mass_matrix.n());
+
+  create_mass_matrix (mapping, dof, quadrature, mass_matrix, function, tmp);
+  //create_right_hand_side (mapping, dof, quadrature, function, tmp);
+  //printf ("RHS created\n");
+
+  constraints.condense (mass_matrix);
+  constraints.condense (tmp);
+  if (boundary_values.size() != 0)
+    MatrixTools::apply_boundary_values (boundary_values,
+                                        mass_matrix, vec, tmp,
+                                        true);
+
+  SolverControl           control(1000,1e-16);
+  PrimitiveVectorMemory<> memory;
+  SolverCG<>              cg(control,memory);
+
+  PreconditionSSOR<> prec;
+  prec.initialize(mass_matrix, 1.2);
+  // solve
+  cg.solve (mass_matrix, vec, tmp, prec);
+
+  // distribute solution
+  constraints.distribute (vec);
+}
+
+
+
+int main (int /*argc*/, char **/*argv*/)
+{
+  std::ofstream logfile ("output");
+  deallog << std::setprecision(2);
+  deallog.attach(logfile);
+  deallog.threshold_double(1.e-10);
+
+
+  Triangulation<3> tria_test;
+  GridGenerator::hyper_cube(tria_test);
+
+
+  for (Triangulation<3>::active_cell_iterator cell = tria_test.begin_active();
+       cell != tria_test.end(); ++cell)
+    {
+      deallog << "Cell " << cell << std::endl;
+      for (unsigned int v=0; v<4; ++v)
+        deallog << "    " << cell->vertex(v) << std::endl;
+    }
+
+
+  FE_ABF<3> fe (0);
+  deallog << "Dofs/cell " << fe.dofs_per_cell
+          << "Dofs/face " << fe.dofs_per_face << std::endl;
+
+  DoFHandler<3> dof_handler (tria_test);
+  dof_handler.distribute_dofs (fe);
+
+  deallog << "Dofs total " << dof_handler.n_dofs () << std::endl;
+
+  Vector<double> solution(dof_handler.n_dofs ());
+  solution = 1;
+
+  // Project solution onto FE field
+  ConstraintMatrix     hn_constraints;
+  hn_constraints.clear ();
+  DoFTools::make_hanging_node_constraints (dof_handler,
+                                           hn_constraints);
+  hn_constraints.close ();
+  MappingQGeneric<3> map_default(1);
+  project (map_default, dof_handler, hn_constraints,
+           QGauss<3> (6), ConstantFunction<3>(1., 3),
+           solution);
+
+  EvaluateDerivative (dof_handler, solution);
+  solution.print (deallog);
+
+  DataOut<3> data_out;
+  data_out.attach_dof_handler (dof_handler);
+  data_out.add_data_vector (solution, "solution");
+  data_out.build_patches (4);
+
+  data_out.write_gnuplot (deallog.get_file_stream());
```

Comment:
I didn't check anything -- I only made sure that it actually executes, which it didn't do before (there were at least 2 places where we got exceptions, see the PRs linked to from #2785).

That said, the test does output the L2 error of the L2 projection of a function we ought to be able to represent, and that output is zero.